containers/cache/ModelsProvider.js

```diff
@@ -14,19 +14,19 @@ const ModelsProvider = ({ children, loginData = {} }) => {
     const api = useApi();
     const eventManagerRef = useRef();
     const cacheRef = useRef();
-
     const [{ loading, error }, setState] = useState({ loading: true });
 
     useEffect(() => {
         const abortController = new AbortController();
         const apiWithAbort = (config) => api({ ...config, signal: abortController.signal });
 
         const setup = async () => {
-            const [user, eventID, userSettingsModel, mailSettingsModel] = await Promise.all([
+            const userSettingsModel = await UserSettingsModel.get(api);
+            const [user, eventID, mailSettingsModel] = await Promise.all([
                 loginData.user || UserModel.get(api),
                 loginData.eventID || getEventID(api),
-                UserSettingsModel.get(api),
                 MailSettingsModel.get(api)
+                // loadLocale(config, userSettingsModel.Locale)
```

Comment:
This shouldn't be here anymore

For the initial loading spinner we need to come up with another solution

I think we need two things: 
* one initial loading state that just does the user request to check if the user is logged in or not
* and if logged in, a second loading state that resolves whatever the application needs

Reply:
> one initial loading state that just does the user request to check if the user is logged in or not

That's why we use the provider. If logged-in or not doesn't really matter.

Reply:
What happens when the user logins with a new locale and the request to fetch the locale takes a long time?

If it displays the UI before the request to fetch the locale has finished I think it's better to show a loading spinner. However this is something we need to add for the dynamic models anyway, and it can be done in a later step.